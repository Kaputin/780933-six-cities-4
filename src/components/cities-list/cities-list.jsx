import React from "react";
import PropTypes from "prop-types";
import {CityPropTypes} from "../../propTypes.js";

export const CitiesList = ({cities, selectedCity, onCityTitleClick}) => {

  return (
    <ul className="locations__list tabs__list">
      {cities.map((city) => (
        <li key={city.id} className="locations__item">
          <a className={`locations__item-link ${city.title === selectedCity.title ? `tabs__item--active` : ``} tabs__item`}>
            <span
              onClick={() => {
                onCityTitleClick(city);
              }}>
              {city.title}
            </span>
          </a>
        </li>
      ))}
    </ul>
  );
};

CitiesList.propTypes = {
  cities: PropTypes.arrayOf(CityPropTypes).isRequired,
  selectedCity: CityPropTypes.isRequired,
  onCityTitleClick: PropTypes.func.isRequired,
};
