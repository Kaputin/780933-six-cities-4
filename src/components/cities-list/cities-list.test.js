import React from "react";
import renderer from "react-test-renderer";
import {CitiesList} from "./cities-list.jsx";

const cities = [
  {
    id: 1,
    title: `Amsterdam`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 2,
    title: `Paris`,
    coordinates: [48.85341, 2.3488]
  },
  {
    id: 3,
    title: `Cologne`,
    coordinates: [50.93333, 6.95]
  },
  {
    id: 4,
    title: `Brussels`,
    coordinates: [50.85045, 4.34878]
  },
  {
    id: 5,
    title: `Hamburg`,
    coordinates: [53.57532, 10.01534]
  },
  {
    id: 6,
    title: `Dusseldorf`,
    coordinates: [51.1722, 6.46]
  }
];

const selectedCity = {
  id: 1,
  title: `Amsterdam`,
  coordinates: [52.38333, 4.9]
};

it(`Should OffersList render correctly`, () => {
  const tree = renderer
    .create(<CitiesList
      cities={cities}
      selectedCity={selectedCity}
      onCityTitleClick={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
