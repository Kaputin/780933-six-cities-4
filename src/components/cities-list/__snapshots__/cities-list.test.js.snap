// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CitysList render correctly 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item--active tabs__item"
    >
      <span
        onClick={[Function]}
      >
        Amsterdam
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link  tabs__item"
    >
      <span
        onClick={[Function]}
      >
        Paris
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link  tabs__item"
    >
      <span
        onClick={[Function]}
      >
        Cologne
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link  tabs__item"
    >
      <span
        onClick={[Function]}
      >
        Brussels
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link  tabs__item"
    >
      <span
        onClick={[Function]}
      >
        Hamburg
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link  tabs__item"
    >
      <span
        onClick={[Function]}
      >
        Dusseldorf
      </span>
    </a>
  </li>
</ul>
`;
